### ISSUE TITLE
3.6.6-RC1 has outdated plugins

### ISSUE THREAD

**Issue by bramley** [phpList/phplist3#821]
The content of the plugins directory looks to be out of date because the files are dated in April.
In particular this isn't the current version of Common plugin.

```
duncan@duncan-OptiPlex-990 ~/Downloads/phplist-3.6.6-RC1/public_html/lists/admin/plugins $ ls -l
total 172
-rw-r--r-- 1 duncan duncan  4396 Apr 29  2021 campaignslicer.php
drwxr-xr-x 2 duncan duncan  4096 Apr 28  2021 CaptchaPlugin
-rw-r--r-- 1 duncan duncan  8743 Apr 28  2021 CaptchaPlugin.php
drwxr-xr-x 3 duncan duncan  4096 Apr 28  2021 CKEditorPlugin
-rw-r--r-- 1 duncan duncan 16003 Apr 28  2021 CKEditorPlugin.php
drwxr-xr-x 3 duncan duncan  4096 Apr 28  2021 Common
drwxr-xr-x 7 duncan duncan  4096 Apr 28  2021 CommonPlugin
-rw-r--r-- 1 duncan duncan  6734 Apr 28  2021 CommonPlugin.php
-rw-r--r-- 1 duncan duncan 35146 Apr 23  2021 COPYING.txt
-rw-r--r-- 1 duncan duncan  4886 Apr 29  2021 dateplaceholder.php
-rw-r--r-- 1 duncan duncan  9924 Apr 29  2021 disposablemailblock.php
-rw-r--r-- 1 duncan duncan 12710 Apr 29  2021 domainthrottlemap.php
-rw-r--r-- 1 duncan duncan  2547 Apr 29  2021 embedremoteimages.php
-rw-r--r-- 1 duncan duncan  3983 Apr 20  2021 inviteplugin.php
drwxr-xr-x 4 duncan duncan  4096 Apr 28  2021 SegmentPlugin
-rw-r--r-- 1 duncan duncan 18421 Apr 28  2021 SegmentPlugin.php
-rw-r--r-- 1 duncan duncan  5397 Apr 23  2021 subjectLinePlaceholdersPlugin.php
```

**Comment by michield** [comment]
Ok, interesting. We should resolve that for the next RC @marianaballa

**Comment by michield** [comment]
This should fix that https://github.com/phpList/phplist3/commit/320995e460685caecfe2026d9d51c09517750175

but we should review this process and make it simpler. @bramley if you can tag your plugins as "versions", we can tell composer to always use the latest stable version. As it is now, it will always take the top of the master branch, which may not always be desirable.

**Comment by michield** [comment]
Hmm, that breaks PHP < 7.4 

I'll update just the plugins, and not the other dependencies

**Comment by michield** [comment]
#822 should do it

**Comment by bramley** [comment]
> if you can tag your plugins as "versions", we can tell composer to always use the latest stable version. As it is now, it will always take the top of the master branch, which may not always be desirable.

The plugins already have tags, e,g, Common Plugin is "3.15.8". But using the master branch will be ok because that will always be the latest tag. I don't apply individual commits to GitHub, instead do everything locally then push the complete new version of a plugin with a tag.

**Comment by bramley** [comment]
This isn't fixed in 3.6.6-RC2 which still has all plugin files dated in April and the older version of Common Plugin.

```
duncan@duncan-OptiPlex-990 ~/Downloads/phplist-3.6.6-RC2/public_html/lists/admin/plugins $ ls -l
total 172
-rw-r--r-- 1 duncan duncan  4396 Apr 29  2021 campaignslicer.php
drwxr-xr-x 2 duncan duncan  4096 Apr 28  2021 CaptchaPlugin
-rw-r--r-- 1 duncan duncan  8743 Apr 28  2021 CaptchaPlugin.php
drwxr-xr-x 3 duncan duncan  4096 Apr 28  2021 CKEditorPlugin
-rw-r--r-- 1 duncan duncan 16003 Apr 28  2021 CKEditorPlugin.php
drwxr-xr-x 3 duncan duncan  4096 Apr 28  2021 Common
drwxr-xr-x 7 duncan duncan  4096 Apr 28  2021 CommonPlugin
-rw-r--r-- 1 duncan duncan  6734 Apr 28  2021 CommonPlugin.php
-rw-r--r-- 1 duncan duncan 35146 Apr 23  2021 COPYING.txt
-rw-r--r-- 1 duncan duncan  4886 Apr 29  2021 dateplaceholder.php
-rw-r--r-- 1 duncan duncan  9924 Apr 29  2021 disposablemailblock.php
-rw-r--r-- 1 duncan duncan 12710 Apr 29  2021 domainthrottlemap.php
-rw-r--r-- 1 duncan duncan  2547 Apr 29  2021 embedremoteimages.php
-rw-r--r-- 1 duncan duncan  3983 Apr 20  2021 inviteplugin.php
drwxr-xr-x 4 duncan duncan  4096 Apr 28  2021 SegmentPlugin
-rw-r--r-- 1 duncan duncan 18421 Apr 28  2021 SegmentPlugin.php
-rw-r--r-- 1 duncan duncan  5397 Apr 23  2021 subjectLinePlaceholdersPlugin.php
```

**Comment by michield** [comment]
Hmm, yes, it takes an older version: https://github.com/phpList/phplist3/runs/4264046587?check_suite_focus=true#step:9:84

I suspect it's because of the Caching, so I'll try switching that off

https://github.com/phpList/phplist3/runs/4264046587?check_suite_focus=true#step:8:12

That will probably fix #825  as well

**Comment by michield** [comment]
Yes, that fixed it: https://github.com/phpList/phplist3/runs/4274334361?check_suite_focus=true#step:6:151

Sorry, I should have checked after my last update
